### PR TITLE
test(arb-adapter): add unit tests

### PR DESCRIPTION
**Pull request by dohaki**
Closes ACX-2207

Uses https://github.com/across-protocol/relayer/pull/1631 as base

### CODE REVIEW

**Comment by bmzig** [comment]
Do you think you could add tests for USDC? Once we use this with the generic adapter, Arbitrum will use CCTP and its canonical bridge, so I'm thinking that the test should cover both bridges.

**Comment by dohaki** [comment]
> Do you think you could add tests for USDC? Once we use this with the generic adapter, Arbitrum will use CCTP and its canonical bridge, so I'm thinking that the test should cover both bridges.

yea good point. I will add them

**Review by james-a-morris — COMMENTED** [review]
Any chance we can add some fuzz testing so that we're not accidentally testing any specific size ranges for deposits? I.e. CCTP had an issue that was uncovered because we tried to bridge too many funds within one message.

File: src/clients/bridges/ArbitrumAdapter.ts [hunk 1]
```diff
@@ -236,10 +267,14 @@ export class ArbitrumAdapter extends CCTPAdapter {
...
     threshold: BigNumber,
     target: BigNumber,
-    simMode = false
+    simMode = false,
+    skipChainIdCheck = false
   ): Promise<TransactionResponse | null> {
     const { chainId } = this;
-    assert(42161 === chainId, `chainId ${chainId} is not supported`);
+
+    if (!skipChainIdCheck) {
```
Comment (bmzig) [inline src/clients/bridges/ArbitrumAdapter.ts:275]: What's the purpose of `skipChainIdCheck`?

**Comment by bmzig** [comment]
Nice. This also LGTM. I'm curious to see how easily this will translate to the generic stuff since it looks like you already are using `getL1DepositInitiatedEvents/getOutstandingCrossChainTransfers` which is super helpful.

**Comment by dohaki** [comment]
> Any chance we can add some fuzz testing so that we're not accidentally testing any specific size ranges for deposits? I.e. CCTP had an issue that was uncovered because we tried to bridge too many funds within one message.

@james-a-morris can you give a rough outline how you would do that in the context of these unit tests that are added as part of this PR? I agree that fuzz testing for the bug you mentioned makes sense but not sure how to achieve that with mocked CCTP contracts... And if that's in the same scope as these unit tests

File: src/clients/bridges/ArbitrumAdapter.ts (hunk 1 above)
Comment (dohaki) [reply src/clients/bridges/ArbitrumAdapter.ts]: Ah yea that's not needed. Removed

**Review by pxrl — APPROVED** [review]

**Comment by dohaki** [comment]
> > Any chance we can add some fuzz testing so that we're not accidentally testing any specific size ranges for deposits? I.e. CCTP had an issue that was uncovered because we tried to bridge too many funds within one message.
> 
> @james-a-morris can you give a rough outline how you would do that in the context of these unit tests that are added as part of this PR? I agree that fuzz testing for the bug you mentioned makes sense but not sure how to achieve that with mocked CCTP contracts... And if that's in the same scope as these unit tests

@james-a-morris I will scope this out of this PR. And if we decide to add those tests, will do so in a separate PR

**Review by james-a-morris — APPROVED** [review]

**Pull request merged**